### ISSUE TITLE
celery 3.1.18 doesn't work w/ Django dummy db

### ISSUE THREAD

**Issue by dblado** [celery/celery#2663]
I don't use the Django ORM so never had a settings.DATABASES configured. After attempting to update from 3.1.17 to 3.1.18 I see an error about improperly configured DATABASES so I added :+1: 

```
DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.dummy'
    }
}
```

Error changed to:

```
www$ celery -A vendop worker
[2015-06-17 22:04:18,755: WARNING/MainProcess] /Users/dblado/vendop/virtenv/lib/python2.7/site-packages/celery/apps/worker.py:161: CDeprecationWarning: 
Starting from version 3.2 Celery will refuse to accept pickle by default.

The pickle serializer is a security concern as it may give attackers
the ability to execute any command.  It's important to secure
your broker from unauthorized access when using pickle, so we think
that enabling pickle should require a deliberate action and not be
the default choice.

If you depend on pickle then you should set a setting to disable this
warning and to be sure that everything will continue working
when you upgrade to Celery 3.2::

    CELERY_ACCEPT_CONTENT = ['pickle', 'json', 'msgpack', 'yaml']

You must only enable the serializers that you will actually use.


  warnings.warn(CDeprecationWarning(W_PICKLE_DEPRECATED))

 
 -------------- celery@globogym.local v3.1.18 (Cipater)
---- **** ----- 
--- * ***  * -- Darwin-14.3.0-x86_64-i386-64bit
-- * - **** --- 
- ** ---------- [config]
- ** ---------- .> app:         vendop:0x102fe9f50
- ** ---------- .> transport:   amqp://guest:**@localhost:5672//
- ** ---------- .> results:     mongodb://localhost:27017
- *** --- * --- .> concurrency: 8 (prefork)
-- ******* ---- 
--- ***** ----- [queues]
 -------------- .> celery           exchange=celery(direct) key=celery
                

[2015-06-17 22:04:18,777: ERROR/Worker-1] Process Worker-1
Traceback (most recent call last):
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/billiard/process.py", line 292, in _bootstrap
    self.run()
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/billiard/pool.py", line 288, in run
    self.after_fork()
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/billiard/pool.py", line 391, in after_fork
    self.initializer(*self.initargs)
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/celery/concurrency/prefork.py", line 82, in process_initializer
    signals.worker_process_init.send(sender=None)
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/celery/utils/dispatch/signal.py", line 166, in send
    response = receiver(signal=self, sender=sender, **named)
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/celery/fixups/django.py", line 214, in on_worker_process_init
    self._close_database()
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/celery/fixups/django.py", line 251, in _close_database
    close()
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/django/db/backends/__init__.py", line 199, in close
    self.set_clean()
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/django/db/backends/dummy/base.py", line 15, in complain
    raise ImproperlyConfigured("settings.DATABASES is improperly configured. "
ImproperlyConfigured: settings.DATABASES is improperly configured. Please supply the ENGINE value. Check settings documentation for more details.
[2015-06-17 22:04:18,777: ERROR/Worker-2] Process Worker-2
Traceback (most recent call last):
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/billiard/process.py", line 292, in _bootstrap
    self.run()
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/billiard/pool.py", line 288, in run
    self.after_fork()
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/billiard/pool.py", line 391, in after_fork
    self.initializer(*self.initargs)
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/celery/concurrency/prefork.py", line 82, in process_initializer
    signals.worker_process_init.send(sender=None)
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/celery/utils/dispatch/signal.py", line 166, in send
    response = receiver(signal=self, sender=sender, **named)
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/celery/fixups/django.py", line 214, in on_worker_process_init
    self._close_database()
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/celery/fixups/django.py", line 251, in _close_database
    close()
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/django/db/backends/__init__.py", line 199, in close
    self.set_clean()
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/django/db/backends/dummy/base.py", line 15, in complain
    raise ImproperlyConfigured("settings.DATABASES is improperly configured. "
ImproperlyConfigured: settings.DATABASES is improperly configured. Please supply the ENGINE value. Check settings documentation for more details.
[2015-06-17 22:04:18,781: ERROR/Worker-3] Process Worker-3
Traceback (most recent call last):
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/billiard/process.py", line 292, in _bootstrap
    self.run()
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/billiard/pool.py", line 288, in run
    self.after_fork()
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/billiard/pool.py", line 391, in after_fork
    self.initializer(*self.initargs)
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/celery/concurrency/prefork.py", line 82, in process_initializer
    signals.worker_process_init.send(sender=None)
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/celery/utils/dispatch/signal.py", line 166, in send
    response = receiver(signal=self, sender=sender, **named)
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/celery/fixups/django.py", line 214, in on_worker_process_init
    self._close_database()
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/celery/fixups/django.py", line 251, in _close_database
    close()
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/django/db/backends/__init__.py", line 199, in close
    self.set_clean()
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/django/db/backends/dummy/base.py", line 15, in complain
    raise ImproperlyConfigured("settings.DATABASES is improperly configured. "
ImproperlyConfigured: settings.DATABASES is improperly configured. Please supply the ENGINE value. Check settings documentation for more details.
[2015-06-17 22:04:18,781: ERROR/Worker-4] Process Worker-4
Traceback (most recent call last):
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/billiard/process.py", line 292, in _bootstrap
    self.run()
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/billiard/pool.py", line 288, in run
    self.after_fork()
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/billiard/pool.py", line 391, in after_fork
    self.initializer(*self.initargs)
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/celery/concurrency/prefork.py", line 82, in process_initializer
    signals.worker_process_init.send(sender=None)
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/celery/utils/dispatch/signal.py", line 166, in send
    response = receiver(signal=self, sender=sender, **named)
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/celery/fixups/django.py", line 214, in on_worker_process_init
    self._close_database()
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/celery/fixups/django.py", line 251, in _close_database
    close()
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/django/db/backends/__init__.py", line 199, in close
    self.set_clean()
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/django/db/backends/dummy/base.py", line 15, in complain
    raise ImproperlyConfigured("settings.DATABASES is improperly configured. "
ImproperlyConfigured: settings.DATABASES is improperly configured. Please supply the ENGINE value. Check settings documentation for more details.
[2015-06-17 22:04:18,784: ERROR/Worker-5] Process Worker-5
Traceback (most recent call last):
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/billiard/process.py", line 292, in _bootstrap
    self.run()
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/billiard/pool.py", line 288, in run
    self.after_fork()
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/billiard/pool.py", line 391, in after_fork
    self.initializer(*self.initargs)
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/celery/concurrency/prefork.py", line 82, in process_initializer
    signals.worker_process_init.send(sender=None)
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/celery/utils/dispatch/signal.py", line 166, in send
    response = receiver(signal=self, sender=sender, **named)
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/celery/fixups/django.py", line 214, in on_worker_process_init
    self._close_database()
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/celery/fixups/django.py", line 251, in _close_database
    close()
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/django/db/backends/__init__.py", line 199, in close
    self.set_clean()
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/django/db/backends/dummy/base.py", line 15, in complain
    raise ImproperlyConfigured("settings.DATABASES is improperly configured. "
ImproperlyConfigured: settings.DATABASES is improperly configured. Please supply the ENGINE value. Check settings documentation for more details.
[2015-06-17 22:04:18,785: ERROR/Worker-6] Process Worker-6
Traceback (most recent call last):
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/billiard/process.py", line 292, in _bootstrap
    self.run()
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/billiard/pool.py", line 288, in run
    self.after_fork()
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/billiard/pool.py", line 391, in after_fork
    self.initializer(*self.initargs)
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/celery/concurrency/prefork.py", line 82, in process_initializer
    signals.worker_process_init.send(sender=None)
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/celery/utils/dispatch/signal.py", line 166, in send
    response = receiver(signal=self, sender=sender, **named)
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/celery/fixups/django.py", line 214, in on_worker_process_init
    self._close_database()
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/celery/fixups/django.py", line 251, in _close_database
    close()
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/django/db/backends/__init__.py", line 199, in close
    self.set_clean()
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/django/db/backends/dummy/base.py", line 15, in complain
    raise ImproperlyConfigured("settings.DATABASES is improperly configured. "
ImproperlyConfigured: settings.DATABASES is improperly configured. Please supply the ENGINE value. Check settings documentation for more details.
[2015-06-17 22:04:18,787: ERROR/Worker-7] Process Worker-7
Traceback (most recent call last):
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/billiard/process.py", line 292, in _bootstrap
    self.run()
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/billiard/pool.py", line 288, in run
    self.after_fork()
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/billiard/pool.py", line 391, in after_fork
    self.initializer(*self.initargs)
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/celery/concurrency/prefork.py", line 82, in process_initializer
    signals.worker_process_init.send(sender=None)
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/celery/utils/dispatch/signal.py", line 166, in send
    response = receiver(signal=self, sender=sender, **named)
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/celery/fixups/django.py", line 214, in on_worker_process_init
    self._close_database()
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/celery/fixups/django.py", line 251, in _close_database
    close()
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/django/db/backends/__init__.py", line 199, in close
    self.set_clean()
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/django/db/backends/dummy/base.py", line 15, in complain
    raise ImproperlyConfigured("settings.DATABASES is improperly configured. "
ImproperlyConfigured: settings.DATABASES is improperly configured. Please supply the ENGINE value. Check settings documentation for more details.
[2015-06-17 22:04:18,789: ERROR/Worker-8] Process Worker-8
Traceback (most recent call last):
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/billiard/process.py", line 292, in _bootstrap
    self.run()
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/billiard/pool.py", line 288, in run
    self.after_fork()
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/billiard/pool.py", line 391, in after_fork
    self.initializer(*self.initargs)
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/celery/concurrency/prefork.py", line 82, in process_initializer
    signals.worker_process_init.send(sender=None)
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/celery/utils/dispatch/signal.py", line 166, in send
    response = receiver(signal=self, sender=sender, **named)
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/celery/fixups/django.py", line 214, in on_worker_process_init
    self._close_database()
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/celery/fixups/django.py", line 251, in _close_database
    close()
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/django/db/backends/__init__.py", line 199, in close
    self.set_clean()
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/django/db/backends/dummy/base.py", line 15, in complain
    raise ImproperlyConfigured("settings.DATABASES is improperly configured. "
ImproperlyConfigured: settings.DATABASES is improperly configured. Please supply the ENGINE value. Check settings documentation for more details.
[2015-06-17 22:04:19,828: WARNING/MainProcess] celery@globogym.local ready.
[2015-06-17 22:04:19,829: ERROR/MainProcess] Process 'Worker-8' pid:72123 exited with 'exitcode 1'
[2015-06-17 22:04:19,842: ERROR/MainProcess] Process 'Worker-7' pid:72122 exited with 'exitcode 1'
[2015-06-17 22:04:19,853: ERROR/MainProcess] Process 'Worker-6' pid:72121 exited with 'exitcode 1'
[2015-06-17 22:04:19,866: ERROR/MainProcess] Process 'Worker-5' pid:72120 exited with 'exitcode 1'
[2015-06-17 22:04:19,878: ERROR/MainProcess] Process 'Worker-4' pid:72119 exited with 'exitcode 1'
[2015-06-17 22:04:19,891: ERROR/MainProcess] Process 'Worker-3' pid:72118 exited with 'exitcode 1'
[2015-06-17 22:04:19,902: ERROR/MainProcess] Process 'Worker-2' pid:72117 exited with 'exitcode 1'
[2015-06-17 22:04:19,914: ERROR/MainProcess] Process 'Worker-1' pid:72116 exited with 'exitcode 1'
[2015-06-17 22:04:20,025: ERROR/MainProcess] Unrecoverable error: WorkerLostError('Could not start worker processes',)
Traceback (most recent call last):
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/celery/worker/__init__.py", line 206, in start
    self.blueprint.start(self)
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/celery/bootsteps.py", line 123, in start
    step.start(parent)
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/celery/bootsteps.py", line 374, in start
    return self.obj.start()
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/celery/worker/consumer.py", line 278, in start
    blueprint.start(self)
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/celery/bootsteps.py", line 123, in start
    step.start(parent)
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/celery/worker/consumer.py", line 821, in start
    c.loop(*c.loop_args())
  File "/Users/dblado/vendop/virtenv/lib/python2.7/site-packages/celery/worker/loops.py", line 48, in asynloop
    raise WorkerLostError('Could not start worker processes')
WorkerLostError: Could not start worker processes
```

**Comment by thedrow** [comment]
Do you configure Celery from the Django settings module?

**Comment by dblado** [comment]
No I have a separate module that sets up celery but it does use Django settings.
FWIW, I am using Celery 1.1.16 in production without any issues. But I believe 1.1.18 now requires  Billiard

```
from __future__ import absolute_import
from celery import Celery
from django.conf import settings
import os

CELERY_MONGODB_BACKEND_SETTINGS = {
    'database': 'vendop_celery',
    'taskmeta_collection': 'my_taskmeta_collection',
}
CELERY_EVENT_QUEUE_TTL = 10
CELERY_RESULT_BACKEND = 'mongodb://{}:{}'.format(settings.DB_HOSTS[0]['host'], settings.DB_HOSTS[0]['port'])

CELERY_SEND_TASK_ERROR_EMAILS = True
ADMINS = settings.ADMINS
SERVER_EMAIL = 'celery@vendop.com'
EMAIL_HOST = settings.EMAIL_HOST
EMAIL_PORT = settings.EMAIL_PORT
EMAIL_HOST_USER = settings.EMAIL_HOST_USER
EMAIL_HOST_PASSWORD = settings.EMAIL_HOST_PASSWORD
EMAIL_USE_TLS = settings.EMAIL_USE_TLS

app = Celery('vendop', backend=CELERY_RESULT_BACKEND, broker=settings.AMQP_BROKER)

app.config_from_object('django.conf:settings')
app.autodiscover_tasks(lambda: settings.INSTALLED_APPS)
```

**Comment by thedrow** [comment]
Celery 1.1 is really old. You should upgrade.

**Comment by dblado** [comment]
oop yes corrected to Celery 3.1.16/18

**Comment by ask** [comment]
Closing this, as we don't have the resources to complete this task.
